### PR TITLE
New Transport for Xmatters and hiding secrets in other Transports

### PR DESCRIPTION
**Pull request by LoveSkylark**
Please give a short description what your pull request is for

1. Add new transport for Xmatters
2. Hide tokens, keys, secrets and passwords in all Transports

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by ottorei** [comment]
Could you please separate these into two different PRs? Thanks.

File: LibreNMS/Alert/Transport/Xmatters.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            ->post($url, $data);
+
+        if ($res->successful()) {
+            return true;
+        }
+
+        throw new AlertTransportDeliveryException($alert_data, $res->status(), $res->body(), $alert_data['msg'], $data);
+    }
+
+    protected function mapSeverityToPriority(string $severity): string
```
Comment (ottorei) [inline LibreNMS/Alert/Transport/Xmatters.php:52]: I think you could probably use Enums here. See an example on: https://github.com/librenms/librenms/blob/298d217a8b4403196ee1d62d1a2c41637ad66342/LibreNMS/Snmptrap/Handlers/VeeamTrap.php#L32

File: LibreNMS/Alert/Transport/Xmatters.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            case 'warning':
+                return 'Medium';
+            case 'ok':
+                return 'Low';
+            default:
+                return 'invalid severity';
+        }
+    }
+
+    protected function mapStateToEventState(int $state): string
```
Comment (ottorei) [inline LibreNMS/Alert/Transport/Xmatters.php:66]: Same here

File: LibreNMS/Alert/Transport/Xmatters.php (hunk 1 above)
Comment (LoveSkylark) [reply LibreNMS/Alert/Transport/Xmatters.php:52]: not sure I understand, Xmatters expects certain values in the json payload "event_state", 
I am translating this from alert_data['state'], if I convert it to enums would not then translate those to a json payload to get an Xmatters readable output?

**Pull request closed without merging**